tests/pyfunc/test_scoring_server.py

```diff
@@ -244,6 +270,47 @@ def test_split_oriented_json_to_df():
     assert set(str(dt) for dt in df.dtypes) == {'object', 'float64', 'int64'}
 
 
+def test_parse_with_schema(pandas_df_with_all_types):
+    # test that datatype for "zip" column is not converted to "int64"
+    # split orientet
+    schema = Schema([ColSpec(c, c) for c in pandas_df_with_all_types.columns])
+    df = _shuffle_pdf(pandas_df_with_all_types)
+    json_str = json.dumps(df.to_dict(orient="split"), cls=NumpyEncoder)
+    df = pyfunc_scoring_server.parse_json_input(json_str,
+                                                orient="split", schema=schema)
+    json_str = json.dumps(df.to_dict(orient="records"), cls=NumpyEncoder)
+    df = pyfunc_scoring_server.parse_json_input(json_str,
+                                                orient="records", schema=schema)
+    assert schema == infer_signature(df[schema.column_names()]).inputs
+
+
+def test_serving_model_with_schema(pandas_df_with_all_types):
+    class TestModel(PythonModel):
+        def predict(self, context, model_input):
+            return [[k, str(v)] for k, v in model_input.dtypes.items()]
+
+    schema = Schema([ColSpec(c, c) for c in pandas_df_with_all_types.columns])
+    df = _shuffle_pdf(pandas_df_with_all_types)
+    with TempDir(chdr=True):
+        with mlflow.start_run() as run:
+            mlflow.pyfunc.log_model("model", python_model=TestModel(),
+                                    signature=ModelSignature(schema))
+        response = pyfunc_serve_and_score_model(
+            model_uri="runs:/{}/model".format(run.info.run_id),
+            data=json.dumps(df.to_dict(orient="split"), cls=NumpyEncoder),
```

Comment:
Mm, this test is a bit weird since we're starting with a Pandas DF. Could we at least include ones where the input is JSON so we can make it obvious how we deal with converting types? E.g., what happens if I specify

9007199254740992

for something that specifies float as its data type? This is valid as a double or as a long, but not as an int or a float.

Reply:
Makes sense. I'll add a test.

Reply:
Actually, it looks like pandas is happy parsing 9007199254740992 as float into 9.007199e+15 which contradicts our rules about only upcasts is ok. I don't think we can do much about it though.  

It's  even worse for integers as it parses as 0 ( I think this changes with pandas version). Again, I don't think it makes too much sense to do something about it for now. 

So, I could add the test, but it would fail.

Reply:
Add the test please, as documentation for the current behavior. You can note that it is weird and subject to change.

Reply:
kk will do

Reply:
Added a test above (testing only the parse in scoring server itself).